src/Options.php

```diff
@@ -1342,4 +1166,37 @@ private function validateContextLinesOption(?int $contextLines): bool
     {
         return $contextLines === null || $contextLines >= 0;
     }
+
+    /**
+     * Calls the resolve method of the internal resolver with a logger so that
+     * validation failures can be logged and investigated.
+     *
+     * @param array<string, mixed> $options
+     *
+     * @return array<string, mixed>
+     */
+    private function resolveWithLogger(array $options = []): array
+    {
+        /**
+         * @var LoggerInterface $logger
+         */
+        $logger = $options['logger'] ?? $this->getLoggerOrNullLogger();
+
+        return $this->resolver->resolve($options, $logger);
+    }
+
+    /**
+     * Merges the passed options with the current options and resolves them.
+     * The result is stored back onto the class field.
+     *
+     * @param array<string, mixed> $override
+     */
+    private function updateOptions(array $override = []): self
+    {
+        $options = array_merge($this->options, $override);
+
+        $this->options = $this->resolveWithLogger($options);
```

Comment:
My only concern is that this is fairly expensive to do for updating 2 or 3 options. Can we pass the overriden options to the resolver and have it only resolve those (new) keys and merge that with `$this->options` instead? Like:


```suggestion
        $this->options = array_merge($this->options, $this->resolver->resolveOnly($options));
```

I don't know if we use setters internally anywhere but I know users do and keeping this as cheap as possible seems like a good idea.